galley/pkg/runtime/processor.go

```diff
@@ -154,21 +202,15 @@ loop:
 		}
 	}
 
-	p.state.close()
+	p.stateStrategy.Reset()
+	p.serviceEntryStrategy.Reset()
 	close(p.stopped)
 	scope.Debugf("Process.process: Exiting process loop")
 }
 
 func (p *Processor) processEvent(e resource.Event) {
 	scope.Debugf("Incoming source event: %v", e)
 	p.recordEvent()
-
-	if e.Kind == resource.FullSync {
-		scope.Infof("Synchronization is complete, starting distribution.")
-		p.state.onFullSync()
```

Comment:
Instead of removing p.state.onFullSync, please use another mechanism to indicate that the handlers should start distributing. You can explicitly notify them here.

Reply:
done.